lib/win-releases.js

```diff
@@ -49,21 +20,22 @@ function parseRELEASES(content) {
 
             var filename = parts[2];
             var isDelta = filename.indexOf('-full.nupkg') == -1;
-
-            var filenameParts = filename
-                .replace(".nupkg", "")
-                .replace("-delta", "")
-                .replace("-full", "")
-                .split(/\.|-/)
-                .reverse();
-
-            var version = _.chain(filenameParts)
-                .filter(function(x) {
-                    return /^\d+$/.exec(x);
+            var version = _.chain(
+                    filename
+                    .replace(suffixRe, '')
+                    .match(versionRe)
+                )
+                .thru(function(matchResult) {
+                    return matchResult ? matchResult[0] : '';
+                })
+                .replace(prereleaseRe, function(prerelease) {
+                    // NuGet doesn't support dots in prereleases
+                    // https://docs.nuget.org/create/versioning#user-content-prerelease-versions
+                    return prerelease.replace(/\./g, '');
                 })
-                .reverse()
-                .value()
-                .join('.');
+                .value();
+
+                if (!version) throw new Error('Release with no versions info: ' + filename);
```

Comment:
Wrapping this in _.chain is kind of ugly.  But I do like the idea of throwing an actually descriptive error rather than "Cannot read property 0 of null".